Sources/NIOPosix/SelectorGeneric.swift

```diff
@@ -137,21 +198,33 @@ internal class Selector<R: Registration>  {
     // Here we add the stored properties that are used by the specific backends
     #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
     typealias EventType = kevent
+    #elseif SWIFTNIO_USE_IO_URING && os(Linux)
+    typealias EventType = URingEvent
+    var eventFD: CInt = -1 // -1 == we're closed
+    var ring = URing()
+    let multishot = URing.io_uring_use_multishot_poll // if true, we run with streaming multishot polls
+    let pipeCache = Cache<Pipe>(maxSize: 8,
```

Comment:
I'm inclined to drop the support for sendfile emulation in v1 of this patch. I'm suspicious that almost no-one is using it right now _anyway_, so I think it's probably fine if we begin by just throwing runtime errors if people try to use it. We can always make a follow-up PR to add support back in if it turns out to be necessary.

Reply:
Nice, will give us a chance to remove some code.

Reply:
Frankly speaking, in a case if we are going to remove possibility to send a file, then the best approach from my point of view would be to disable a possibility to create a file region parcel on the compile time.
But it will require a conditional compilation, and if you do not want to have many of them in the code, then a runtime error is only a way... not a perfect from my point of view because end user fill find that later...

Reply:
The bigger problem is not the conditional compilation, it's the fact that we don't want to break the API for this. So sadly for now this needs to stay as-is. But it'll be a useful sanity check to find out if anyone actually is sending files.

Reply:
Just a gentle ping that you can delete this code for now.

Reply:
Done within [commit](https://github.com/ordo-one/swift-nio/commit/7fffd46b8a806b0939405d9bc660dab365fc4bd6)